deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,317 @@
+                return per_sample_binary_cross_entropy(dataset.label_col,
+                                                       model.predict_proba(dataset.features_columns))
+
+        train_scores = scoring_func(train_dataset)
+        test_scores = scoring_func(test_dataset)
+
+        # Create and fit model to predict the per sample error
+        error_model = create_error_model(train_dataset, random_state=self.random_state)
+        error_model.fit(train_dataset.features_columns, y=train_scores)
+
+        # Check if fitted model is good enough
+        error_model_predicted = error_model.predict(test_dataset.features_columns)
+        error_model_score = r2_score(test_scores, error_model_predicted)
+
+        # This check should be ignored if no information gained from the error model (low r2_score)
+        if error_model_score < self.min_error_model_score:
+            raise DeepchecksProcessError(f'Unable to train meaningful error model '
+                                         f'(r^2 score: {format_number(error_model_score)})')
+
+        error_fi = calculate_feature_importance(error_model, test_dataset,
+                                                permutation_kwargs={'random_state': self.random_state})
+        error_fi.sort_values(ascending=False, inplace=True)
+
+        n_samples_display = min(self.n_display_samples, len(test_dataset))
+        display_data = test_dataset.data
+        error_col_name = 'Deepchecks model error'
+        display_error = pd.Series(error_model_predicted, name=error_col_name, index=test_dataset.data.index)
+
+        display = []
+        value = {}
+        self._scorer_name = scorer.name
+        weak_color = '#d74949'
+        ok_color = colors['Test']
+
+        for feature in error_fi.keys()[:self.max_features]:
+            if error_fi[feature] < self.min_feature_contribution:
+                break
+
+            data = pd.concat([display_data[feature], display_error], axis=1)
+
+            # Violin plot for categorical features, scatter plot for numerical features
+            if feature in cat_features:
+                # find categories with the weakest performance
+                error_per_segment_ser = data.groupby(feature).agg(['mean', 'count']
+                                                                  )[error_col_name].sort_values('mean', ascending=False)
+                cum_sum_ratio = error_per_segment_ser['count'].cumsum() / error_per_segment_ser['count'].sum()
+
+                # Partition data into two groups - weak and ok:
+
+                in_segment_indicis = cum_sum_ratio <= self.min_segment_size
+                weak_categories = error_per_segment_ser.index[in_segment_indicis]
+                ok_categories = error_per_segment_ser.index[~in_segment_indicis]
+
+                # Calculate score for each group and assign label and color
+                ok_name_feature, segment1_details = get_segment_details(model, scorer, test_dataset, data,
```

Comment:
maybe better name for the segments? like `ok_segment` and `weak_segment`?

Reply:
at this stage I don't know which one is ok and which one is weak